### ISSUE TITLE
Collect compat.sh test cases

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#7094]
Currently, `collect_available_test_cases` in `check_test_cases` only collects entries from unit tests and `ssl-opt.sh`. The goal of this task is to make it collect entries from `compat.sh` as well (for the sake of `analyze_outcomes.py`).

Since `compat.sh` does not have a fixed list of test cases, we need a way to generate it. I see two potential ways to do that:

* Add an option `--list-test-cases` to `compat.sh` which makes it list all potential test cases without executing them. Note that it must work without a build of mbedtls and without openssl/gnutls available. So the idea would be that any runtime check of “is this supported?” is assumed yes, and output a description line instead of running the test case.
* Have Python code that figures out the list of available cipher suites and enumerates them with peer and verify values.

Either way, we may have to add exclusions for cipher suites or other features that are not supported by any version of openssl/gnutls. However, a complete list of exclusions is out of scope of the present task: we'll make follow-up tasks if necessary based on https://github.com/Mbed-TLS/mbedtls/issues/5390.

Prerequisite: https://github.com/Mbed-TLS/mbedtls/issues/5692

Impacts: https://github.com/Mbed-TLS/mbedtls/issues/5390
